cni/network/plugin/main.go

```diff
@@ -52,6 +54,16 @@ func printVersion() {
 	fmt.Printf("Azure CNI Version %v\n", version)
 }
 
+// send error report to hostnetagent if CNI encounters any error.
+func reportPluginError(reportManager *telemetry.ReportManager, tb *telemetry.TelemetryBuffer, err error) {
+	logger.Error("Report plugin error")
+	reflect.ValueOf(reportManager.Report).Elem().FieldByName("ErrorMessage").SetString(err.Error())
```

Comment:
Why do we have to use reflection here?

Reply:
refactored to remove reflection and use the single telemetry package level variable instead